utils/create_dummy_models.py

```diff
@@ -0,0 +1,741 @@
+                if "error" in results[config_name][framework][arch_name]:
+                    if config_name not in failed_results:
+                        failed_results[config_name] = {}
+                    if framework not in failed_results[config_name]:
+                        failed_results[config_name][framework] = {}
+                    if arch_name not in failed_results[config_name][framework]:
+                        failed_results[config_name][framework][arch_name] = {}
+                    error = results[config_name][framework][arch_name]["error"]
+                    failed_results[config_name][framework][arch_name]["error"] = error
+
+    return failed_results
+
+
+def build_simple_report(results):
+
+    text = ""
+    for config_name in results:
+        for framework in FRAMEWORKS:
+            if framework not in results[config_name]:
+                continue
+            for arch_name in results[config_name][framework]:
+                if "error" in results[config_name][framework][arch_name]:
+                    result = results[config_name][framework][arch_name]["error"]
+                else:
+                    result = "OK"
+                text += f"{arch_name}: {result}\n"
+
+    return text
+
+
+if __name__ == "__main__":
+
+    def list_str(values):
+        return values.split(",")
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--all", action="store_true", help="Will create all tiny models.")
+    parser.add_argument(
+        "--no_check",
+        action="store_true",
+        help="If set, will not check the validity of architectures. Use with caution.",
+    )
+    parser.add_argument(
+        "-m",
+        "--model_types",
+        type=list_str,
+        help="Comma-separated list of model type(s) from which the tiny models will be created.",
+    )
+    parser.add_argument(
+        "--black_list",
+        type=list_str,
+        help="Comma-separated list of model type(s) to ignore.",
+        default="convbert,blenderbot-small,rag,dpr,retribert,layoutlmv2",
+    )
+    parser.add_argument("output_path", type=Path, help="Path indicating where to store generated model.")
+
+    args = parser.parse_args()
+
+    if not args.all and not args.model_types:
+        raise ValueError("Please provide at least one model type or pass `--all` to export all architectures.")
+
+    config_classes = CONFIG_MAPPING.values()
+    if not args.all:
+        config_classes = [CONFIG_MAPPING[model_type] for model_type in args.model_types]
+
+    # A map from config classes to tuples of processors (tokenizer, feature extractor, processor) classes
+    processor_type_map = {c: get_processor_types_from_config_class(c) for c in config_classes}
+
+    # Ignore some model types
+    # TODO: Discuss with Lysandre about the reason behind this
+    if args.black_list:
+        config_classes = [c for c in config_classes if c.model_type not in args.black_list]
```

Comment:
It was a blacklist so that I could develop a bit quicker. We don't need to keep it. There were a few models for which the script was not passing so I would add them there to ensure we had a sufficient amount of architectures covered in the least amount of time.